packages/expo-secure-store/android/src/main/java/expo/modules/securestore/AuthenticationHelper.kt

```diff
@@ -0,0 +1,212 @@
+import javax.crypto.Cipher
+import javax.crypto.spec.GCMParameterSpec
+
+class AuthenticationHelper(
+  private val context: Context,
+  private val moduleRegistry: ModuleRegistry
+) {
+  companion object {
+    private const val AUTHENTICATION_PROMPT_PROPERTY = "authenticationPrompt"
+    const val REQUIRE_AUTHENTICATION_PROPERTY = "requireAuthentication"
+  }
+
+  private val uiManager = moduleRegistry.getModule(UIManager::class.java)
+  private var isAuthenticating = false
+
+  // Authentication callback decides whether the operation requires authentication (either by
+  // requiresAuthentication argument, or from options). When item needs to be encrypted/decrypted an
+  // instance of Authentication callback is passed to the relevant method.
+  // The method prepares the cipher and starts authentication callback with it. If the operation
+  // requires authentication, the biometric prompt is shown, otherwise the encryption callback
+  // is called.
+  // When the user is authenticated the encryption callback is ran with the unlocked cipher and does
+  // encryption/decryption. Finally the PostEncryptionCallback may be ran with the object returned
+  // by previous callback (to save encrypted data to SharedPreferences).
+
+  val defaultCallback: AuthenticationCallback = object : AuthenticationCallback {
+    override fun checkAuthentication(
+      promise: Promise,
+      cipher: Cipher,
+      gcmParameterSpec: GCMParameterSpec,
+      options: ReadableArguments,
+      encryptionCallback: EncryptionCallback,
+      postEncryptionCallback: PostEncryptionCallback?
+    ) {
+      val requiresAuthentication = options.getBoolean(REQUIRE_AUTHENTICATION_PROPERTY, false)
+
+      checkAuthentication(
+        promise, requiresAuthentication, cipher, gcmParameterSpec, options, encryptionCallback, postEncryptionCallback
+      )
+    }
+
+    override fun checkAuthentication(
+      promise: Promise,
+      requiresAuthentication: Boolean,
+      cipher: Cipher,
+      gcmParameterSpec: GCMParameterSpec,
+      options: ReadableArguments,
+      encryptionCallback: EncryptionCallback,
+      postEncryptionCallback: PostEncryptionCallback?
+    ) {
+      if (requiresAuthentication) {
+        openAuthenticationPrompt(promise, options, encryptionCallback, cipher, gcmParameterSpec, postEncryptionCallback)
+      } else {
+        handleEncryptionCallback(promise, encryptionCallback, cipher, gcmParameterSpec, postEncryptionCallback)
+      }
+    }
+  }
+
+  fun handleEncryptionCallback(
+    promise: Promise,
+    encryptionCallback: EncryptionCallback,
+    cipher: Cipher,
+    gcmParameterSpec: GCMParameterSpec,
+    postEncryptionCallback: PostEncryptionCallback?
+  ) {
+    try {
+      encryptionCallback.run(promise, cipher, gcmParameterSpec, postEncryptionCallback)
+    } catch (exception: GeneralSecurityException) {
+      Log.w(SecureStoreModule.TAG, exception)
+      promise.reject(
+        "ERR_SECURESTORE_ENCRYPT_FAILURE,
```

Comment:
I believe the code does not compile because of this missing `"` 😅 
```suggestion
        "ERR_SECURESTORE_ENCRYPT_FAILURE",
```